content/smyja/how-to-use-graphql-with-django.md

```diff
@@ -0,0 +1,287 @@
+---
+Title: "How to use GrahQl with Django"
+Description: "A Basic GUide on How to use Graphql with Django"
+DatePublished: "2022-06-05"
+Categories:
+  - "web-development"
+  - "computer-science"
+Tags:
+  - "APIs"
+  - "Django"
+  - "Python"
+CatalogContent:
+  - "how-to-use-graphql-with-django"
+  - "paths/computer-science"
+---
+_**Prerequisites:** Understanding of Python and Django._
+
+_**Versions:** Python 3.8.10, Django 4.0.4_
+
+## Introduction
+GraphQL is a query language for APIs and a runtime for fulfilling those queries with your existing data. Unlike a REST api, Graphql api’s do not require verbs(PUT,POST,GET) for requests, they do not need multiple Endpoints. They have just one endpoint and making a query to that endpoint is all that's needed. 
+
+This tutorial will cover the creation of a CRUD(create,read,update,delete)graphql api for a restaurant with django.
+
+ **Properties of Graphql**:
+
+The following terms would be used a lot when interacting with graphql. It is important that you know them, though we wouldn't cover all of them in this tutorial.
+
+- schema
+- query
+- Nesting
+- Mutation
+- Subscription
+- Resolvers
+
+**Schema:** 
+GraphQl Schema describes the functionality available to the client applications that connect to it. 
+
+**Query:**
+A Query is a type on a schema that defines the kind of operations that can be done to read or fetch data. Queries represent the `GET` request.
+
+**Resolvers:**
+They are functions that return values for fields that exist on types in a schema. 
+
+**Mutation:**
+A mutation is a type on a schema that defines the kind of operations that can be done to modify data.
+
+**Subscription:**
+Subscriptions in graphql are for notifying your client server in real time about changes in the data.
+
+**Nesting:**
+A nested query is a query that is a child of another query.
+
+## Setting up graphql with django
+First, we need to install Django and create a project. I will be using a bash script i created to do this.
+[Django-bash-script](https://github.com/Smyja/Django-bash)
+
+To use graphql with django, you will need to install graphene-django
+
+```pip install graphene-django```
+
+Add it to your settings.py file:
```

Comment:
```suggestion
Add it to your `settings.py` file:

```

Couple points here:

- When referring to files or folders used in the article, surround with single back-ticks.
- Similar to markdown headings, multi-line code snippets should have one empty line above and below them.